### PR TITLE
ctdb: modularize the ctdb.py code a little

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Make it possible for future changes to store the metadata differently.

### CODE REVIEW

File: sambacc/typelets.py [hunk 1]
```diff
@@ -20,8 +20,17 @@
...
 ExcValue = typing.Optional[BaseException]
 ExcTraceback = typing.Optional[TracebackType]
+
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+elif typing.TYPE_CHECKING:
+    from typing_extensions import Self
+else:
+    Self = typing.Any
```
Comment (synarete) [inline sambacc/typelets.py:36]: Nice idea!

When checking python version, I usually use [platform.python_version_tuple](https://docs.python.org/3/library/platform.html#platform.python_version_tuple), but `sys.version_info` is just fine here.

**Review by synarete — APPROVED** [review]
lgtm

**Comment by phlogistonjohn** [comment]
@anoopcs9 PTAL

**Review by anoopcs9 — APPROVED** [review]
I was bit reluctant to review this change with all the language specific idioms used for introducing the new classes. But overall I think the rearrangement looks good.

**Pull request merged**